libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -199,3 +199,39 @@ def _insert_embeddings(
         except DBAPIError as e:
             logging.error(e.__cause__)
         return ids
+
+    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> Optional[bool]:
+        """Delete embeddings in the vectorstore by the ids.
+
+        Args:
+            ids: List of IDs to delete.
+            kwargs: vectorstore specific parameters.
+
+        Returns:
+            Optional[bool]
+        """
+
+        if ids is None or len(ids) == 0:
+            logging.info("Empty list of ids provided")
+            raise ValueError("No ids provided to delete.")
+
+        result = self.delete_texts_by_ids(ids)
+        if result == 0:
+            logging.info("Provided ids not found in the DB.")
+            return False
+
+        logging.info(result, " rows affected.")
+        return True
+
+    def delete_texts_by_ids(self, ids: Optional[List[str]] = None) -> int:
```

Comment:
nit: Let's indicate this is a private function via the function name. That way users are not inclined to directly call this function `_delete_texts_by_ids`, and it follows the existing convention.

Reply:
Okay